spec/acceptance/99_client_server_spec.rb

```diff
@@ -0,0 +1,56 @@
+require 'spec_helper_acceptance'
+
+describe 'yum::server class', if: RSpec.configuration.yum_full do
+  server = only_host_with_role(hosts, 'server')
+  client = only_host_with_role(hosts, 'client')
+  context 'yum::server' do
+    it 'should work with no errors' do
+      pp = <<-EOS
+      include ::yum::server
+      EOS
+
+      # Run it twice and test for idempotency
+      apply_manifest_on(server, pp, :catch_failures => true)
+      apply_manifest_on(server, pp, :catch_changes  => true)
+    end
+
+    it 'should create repo' do
+      on server, 'mkdir -p /opt/repos/cowsay/7'
+      on server, 'wget https://dl.fedoraproject.org/pub/epel/7/x86_64/Packages/c/cowsay-3.04-4.el7.noarch.rpm -O /opt/repos/cowsay/7/cowsay-3.04-4.el7.noarch.rpm'
+      on server, 'cd /opt/repos/cowsay/7 && createrepo -v .'
+    end
+  end
+  context 'yum::repo' do
+    it 'sould work with no errors' do
+      pp = <<-EOS
+      include ::yum
+      yum::repo { 'test':
+        description => 'Test yum-server serving up yum repos',
+        baseurl     => ['http://yum-server/cowsay/\$releasever/'],
+        enabled     => true,
+        gpgcheck    => false,
+      }
+      package { 'cowsay':
+        ensure  => 'installed',
+        require => Yum::Repo['test'],
+      }
+      EOS
+      apply_manifest_on(client, pp, :catch_failures => true)
+      apply_manifest_on(client, pp, :catch_changes  => true)
+    end
+
+    describe yumrepo('test'), :node => client  do
```

Comment:
from a style perspective, should all the tests that run on the client just be inside of an `it` block that has `on client` ?

Reply:
I don't think that will work.  The `on client` idiom is just for executing shell commands on a node.

```
on client, 'run command'
```

If I understand correctly this won't work:

```
on client do
  it 'should do something' do
     test...
  end
end
```

The block of `on` will allow accessing things like stdout

```
on client, 'some command' do
  puts stdout
end
```

The `on` syntax I don't think is intended to dictate tests, just run commands.  The only way I know to use serverspec tests with multiple nodes is the `describe <resource>, :node => client` type pattern.  For example docs for yumrepo: https://serverspec.org/resource_types.html#yumrepo.  I've been using this pattern so long I'm not even sure where I found out about `:node => `, can't seem to find in serverspec advanced docs.

Upstream docs for `on`: https://www.rubydoc.info/github/puppetlabs/beaker/Beaker/DSL/Helpers/HostHelpers#on-instance_method